pkg/operator/endpoints/metrics.go

```diff
@@ -0,0 +1,449 @@
+	if err != nil {
+		return err
+	}
+	if avg != nil {
+		regressionStats.Avg = *avg
+	}
+	if regressionStats.SampleCount > 0 {
+		apiMetrics.RegressionStats = &regressionStats
+	}
+	return nil
+}
+
+func GetAPIDimensions(appName string, api *context.API) []*cloudwatch.Dimension {
+	return []*cloudwatch.Dimension{
+		{
+			Name:  aws.String("AppName"),
+			Value: aws.String(appName),
+		},
+		{
+			Name:  aws.String("APIName"),
+			Value: aws.String(api.Name),
+		},
+		{
+			Name:  aws.String("APIID"),
+			Value: aws.String(api.ID),
+		},
+	}
+}
+
+func GetLatencyMetricsDef(routeName string, period int64) []*cloudwatch.MetricDataQuery {
+	networkDataQueries := []*cloudwatch.MetricDataQuery{
+		{
+			Id:    aws.String("latency"),
+			Label: aws.String("Latency"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: &cloudwatch.Metric{
+					Namespace:  aws.String(config.Cortex.Namespace),
+					MetricName: aws.String("response-time.instance.cortex"),
+					Dimensions: []*cloudwatch.Dimension{
+						{
+							Name:  aws.String("RequestPath"),
+							Value: aws.String(routeName),
+						},
+					},
+				},
+				Stat:   aws.String("Average"),
+				Period: aws.Int64(period),
+			},
+		},
+		{
+			Id:    aws.String("request_count"),
+			Label: aws.String("RequestCount"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: &cloudwatch.Metric{
+					Namespace:  aws.String(config.Cortex.Namespace),
+					MetricName: aws.String("response-time.instance.cortex"),
+					Dimensions: []*cloudwatch.Dimension{
+						{
+							Name:  aws.String("RequestPath"),
+							Value: aws.String(routeName),
+						},
+					},
+				},
+				Stat:   aws.String("SampleCount"),
+				Period: aws.Int64(period),
+			},
+		},
+	}
+
+	return networkDataQueries
+}
+
+func GetRegressionMetricDef(appName string, api *context.API, period int64) []*cloudwatch.MetricDataQuery {
+	metric := &cloudwatch.Metric{
+		Namespace:  aws.String(config.Cortex.Namespace),
+		MetricName: aws.String("Prediction"),
+		Dimensions: GetAPIDimensions(appName, api),
+	}
+
+	regressionMetric := []*cloudwatch.MetricDataQuery{
+		{
+			Id:    aws.String("min"),
+			Label: aws.String("Min"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: metric,
+				Stat:   aws.String("Minimum"),
+				Period: aws.Int64(period),
+			},
+		},
+		{
+			Id:    aws.String("max"),
+			Label: aws.String("Max"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: metric,
+				Stat:   aws.String("Maximum"),
+				Period: aws.Int64(period),
+			},
+		},
+		{
+			Id:    aws.String("avg"),
+			Label: aws.String("Avg"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: metric,
+				Period: aws.Int64(period),
+				Stat:   aws.String("Average"),
+			},
+		},
+	}
+
+	return regressionMetric
+}
+
+func GetNetworkStatsDef(appName string, api *context.API, period int64) []*cloudwatch.MetricDataQuery {
+	dimensions := GetAPIDimensions(appName, api)
+
+	status200 := append([]*cloudwatch.Dimension{}, dimensions...)
```

Comment:
Can you remove this line, and use `dimensions` in the `append()` below instead of `status200`? Same question for 400 and 500